src/ioc.test.ts

```diff
@@ -81,5 +81,127 @@ describe("ioc tests", () => {
     sinon.assert.calledOnceWithExactly(overrideDependency, "in caller");
     sinon.assert.calledOnceWithExactly(defaultDependency, "in caller");
   });
+  it("setDefault full functionality", async () => {
+    // Setup Dependencies
+    const defaultDependency = sinon.stub();
+    const defaultDependencyObject = {mainDep: defaultDependency};
+    const tooManyEntryDependencyObject = {mainDep: defaultDependency, secondDep: defaultDependency};
+    const additionalDependency = sinon.stub();
+    const additionalDependencyObject = {mainDep: additionalDependency};
+
+    // Setup IOC
+    const iocT5 = iocFactory();
+
+    // setDefault() requires only one default to be set at a time
+    assert.throws(() => {iocT5.setDefault(tooManyEntryDependencyObject);}, {name: 'Error', message: `ioc.setDefault({name}) takes only one thing at a time.`});
+
+    // Successful setDefault() returns iocGet wrapper for the given name & get() basic functionality behaves as expected
+    assert.strictEqual(iocT5.setDefault(defaultDependencyObject)(), defaultDependency);
+
+    // Dependency names must be unique
+    assert.throws(() => {iocT5.setDefault(additionalDependencyObject);}, {name: 'Error', message: `ioc.setDefault({mainDep:...}) -> already set.}`});
+  });
+  it("get function error", async () => {
+    // Setup IOC
+    const iocT6 = iocFactory();
+
+    // get() throws error if a default has not yet been set
+    assert.throws(() => {iocT6.get("mainDep");}, {name: 'Error', message: `ioc.get("mainDep") no current or default defined.`});
+  });
+  it("setDeps functionality", async () => {
+    // Setup Dependencies
+    const defaultDependency = sinon.stub();
+    const defaultDependencyObject = {mainDep: defaultDependency};
+    const overrideDependency = sinon.stub();
+    const overrideDependencyObject = {mainDep: overrideDependency};
+
+    // Setup IOC
+    const iocT7 = iocFactory();
+
+    // set() throws error if a default has not yet been set
+    assert.throws(() => {iocT7.setDeps(overrideDependencyObject, true);}, {name: 'Error', message: `ioc.set("mainDep") no default defined.`});
+
+    // set() overwrites a dependency successfully
+    iocT7.setDefault(defaultDependencyObject);
+    assert.strictEqual(iocT7.get("mainDep"), defaultDependency);
+    iocT7.setDeps(overrideDependencyObject, true);
+    assert.strictEqual(iocT7.get("mainDep"), overrideDependency);
+
+    // set() cannot be used in production mode
+    iocT7.productionMode();
+    assert.throws(() => {iocT7.setDeps(defaultDependencyObject);}, {name: 'Error', message: `set("mainDep") is not valid in production mode.`});
+  });
+  it("setDeps functionality", async () => {
```

Comment:
This is a duplicate of the one on line 111. Not a big deal but it'd be ideal to differentiate them a bit.